### PR TITLE
pre-commit: improve test

### PR DESCRIPTION
**Pull request by asottile**
Now covers:
- `PYTHONPATH` based setup
- relative virtualenv symlinks

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

Resolves https://github.com/pre-commit/pre-commit/issues/669
See also https://github.com/Homebrew/homebrew-core/issues/21386

### CODE REVIEW

**Comment by asottile** [comment]
CC @ilovezfs

File: Formula/pre-commit.rb [hunk 1]
```diff
@@ -47,7 +47,16 @@ def post_install
...
       system bin/"pre-commit", "install"
-      system bin/"pre-commit", "run", "--all-files"
+      (testpath/"f").write "hi\n"
+      system "git", "add", "f"
+
+      ENV["GIT_AUTHOR_NAME"] = "test user"
+      ENV["GIT_AUTHOR_EMAIL"] = "test@example.com"
+      ENV["GIT_COMMITTER_NAME"] = "test user"
+      ENV["GIT_COMMITTER_EMAIL"] = "test@example.com"
+      git_exe = `which git`.strip
```
Comment (ilovezfs) [inline Formula/pre-commit.rb]: git_exe = which("git") should work I think

**Comment by ilovezfs** [comment]
@asottile are you watching me? How did you know I was just looking at this PR? hehe

**Comment by asottile** [comment]
👻 spoooooky 👻

File: Formula/pre-commit.rb (hunk 1 above)
Comment (asottile) [reply Formula/pre-commit.rb]: indeed it does! updated!

File: Formula/pre-commit.rb [hunk 2]
```diff
@@ -47,7 +47,16 @@ def post_install
...
       system bin/"pre-commit", "install"
-      system bin/"pre-commit", "run", "--all-files"
+      (testpath/"f").write "hi\n"
+      system "git", "add", "f"
+
+      ENV["GIT_AUTHOR_NAME"] = "test user"
+      ENV["GIT_AUTHOR_EMAIL"] = "test@example.com"
+      ENV["GIT_COMMITTER_NAME"] = "test user"
+      ENV["GIT_COMMITTER_EMAIL"] = "test@example.com"
+      git_exe = which('git')
```
Comment (ilovezfs) [inline Formula/pre-commit.rb]: needs to be " not ' or style won't be happy

Comment (asottile) [reply Formula/pre-commit.rb]: oops! opposite from python lol

Comment (ilovezfs) [reply Formula/pre-commit.rb]: It's just a rubocop setting. Cask still does the single-where-no-interpolating thing https://github.com/caskroom/homebrew-cask/blob/master/Casks/java.rb

Comment (asottile) [reply Formula/pre-commit.rb]: ah makes sense 👍

**Comment by ilovezfs** [comment]
Thanks @asottile! We really appreciate the upstream collaboration :)

**Comment by asottile** [comment]
Thanks!  Glad to have an easy way to get started with `pre-commit` \o/

**Pull request merged**